Fix deployment/publishing page

mlqn opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description

- Implemented the new `status` and `statusDate` properties from [KubernetesWrapper](https://github.com/Altinn/app-kubernetes-wrapper/pull/27) to get the real status of the Kubernetes deployment
- Removed complexity by decoupling Pipeline and Kubernetes statuses as they might sometimes be different (e.g. if a deployment fails and a rollback happens, the Pipeline fails but the KubernetesWrapper succeeds)
  - Removed `deployedInEnv` property from pipeline objects
  - Removed custom `DeploymentStatus` as it was a mix between Pipeline status/result and Kubernetes status
  - Created two separate lists : `pipelineDeploymentList` and `kubernetesDeploymentList`

### Overview page

- Added version and link to the application
- Added inProgress status
- Sorted environments by order of importance to display `Production` and `TT02` first instead of having them at the end of the list
- Refactored the overview page to return correct loading/failed messages (created new `Header` and `Deployments`, renamed `AppEnvironments` to `DeploymentStatusList`, `AppStatus` to `DeploymentStatus`, `AppLogs` to `DeploymentLogList`)

![overview-1-](https://github.com/Altinn/altinn-studio/assets/24462611/4cc20c66-e903-47ce-989a-bae18cec0b3a)

### Deployment/Publishing page

- Reused design from the overview page to display the kubernetes status on the publishing page
- Displayed all builds (not only the succeeded ones) and their status (succeeded, partiallySucceeded, failed, canceled, none/inProgress)
- Displayed link to build log (expired after 1 month)
- Replaced `LegacyTable` with `Table` and `LegacySelect` with `Combobox`
- Displayed the `no version to deploy` message when no version has been built
- Made code and messages consistent between the overview and the publish page
- Added missing isPending and isError from mutations (https://github.com/Altinn/altinn-studio/issues/12126)
- Sorted environments by order of importance to display Production and TT02 first instead of having them at the end of the list
- Removed `appDeploymentSagas` and some duplicated types
- Added missing tests and removed use of queryClientMock in some tests (https://github.com/Altinn/altinn-studio/issues/12463)
- Fixed strange bouncing effect (https://github.com/Altinn/altinn-studio/issues/11913)
- Added `title` to `AltinnContentLoader` svg

![succeeded (1)](https://github.com/Altinn/altinn-studio/assets/24462611/52265f12-26ce-4553-b162-43ac3f566b06)

### Backend

- Added `?labelSelector=release={org}-{app}` in kubernetes url to only return data for the selected app
- Decoupled Pipeline deployment and Kubernetes deployment by creating a new KubernetesDeploymentsController and a new KubernetesDeploymentsService

## Related Issue(s)
- #10312
- #12206
- #11624
- #11960
- #11659
- #12128
- #12126
- #11548
- #11913
- #11359

## Verification

- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
- [x] Relevant automated test added (if you find this hard, leave it and we'll help out)

nkylstad commented on frontend/app-development/features/appPublish/components/DeployDropdown.tsx:
```diff
@@ -0,0 +1,114 @@
+    .filter((image) => image.build.result === BuildResult.succeeded)
+    .map((image) => ({
+      value: image.tagName,
+      label: `Version ${image.tagName} (${formatDateTime(image.created)})`,
```
Could we move this text into the texts file to make it translatable?

mlqn replied:
Done 👍

nkylstad reviewed (commented):
I have only gotten through about one third so far 😅 But so far this looks amazing! I added one small comment on a text that could be translated, will continue to look thought tonight or tomorrow 😊

mirkoSekulic commented:
Great improvement.

nkylstad commented on frontend/packages/shared/src/pure/date-format.ts:
May be out of scope for this PR, but should we move these functions to the `libs/studio-pure-functions` library?

mlqn replied:
I agree, it seems that we can move the whole `pure` folder to `libs/studio-pure-functions`.
So I moved the date-format functions to the `libs/studio-pure-functions` library in this PR since they are only used by the deployment/publishing page
And I created a new PR to move the utils functions: https://github.com/Altinn/altinn-studio/pull/12523

nkylstad reviewed (approved):
Overall a fantastic piece of work! I don't really have anything big to comment on, only a couple of tiny details. 
One thing I did notice while testing the branch locally is that I do not get any mocked status from KubernetesWrapper. This might be something for another PR, but we do have a `studio-azure-mock`, could we also create a `studio-kubernetes-wrapper-mock` to mock some data for app statuses for some test apps?

nkylstad commented on frontend/language/src/nb.json:
```diff
@@ -89,6 +89,7 @@
   "app_deployment.table.status": "Status",
   "app_deployment.table.version_col": "Versjon",
   "app_deployment.technical_error_1": "Huff da, vi opplever en teknisk feil og får derfor ikke deployet applikasjonen din. Gjør en ny deploy eller forsøk igjen senere. Dersom problemet vedvarer, <a>ta kontakt med oss</a>.",
+  "app_deployment.version_label": "Version {{tagName}} ({{createdDateTime}})",
```
```suggestion
  "app_deployment.version_label": "Versjon {{tagName}} ({{createdDateTime}})",
```

mlqn replied:
Done 👍

mlqn commented on frontend/app-development/features/appPublish/containers/DeploymentContainer.tsx:
```diff
@@ -0,0 +1,94 @@
+          <DeploymentEnvironment
+            key={index}
+            envName={env.name}
+            isProduction={env.type.toLowerCase() === 'production'}
```
Done 👍

nkylstad reviewed (commented):
We should probably create some e2e-tests for the updated deploy-page as well. I have added a separate issue for that, to avoid bloating this PR 😅 https://github.com/Altinn/altinn-studio/issues/12555

With regards to the KubernetesWrapper mock, if that is a big thing then we could consider creating a separate issue/PR for that as well.

mlqn commented:
Here are the last changes I made :

- Removed the code that was retrieving the status from KubernetesWrapper since we can now rely on the status of the pipeline by using the `--atomic` flag

- Fixed the KubernetesWrapper mock, so we can now simulate a deployment locally

https://github.com/Altinn/altinn-studio/assets/24462611/c294ad21-3459-43d6-a2de-58750fede35f

- Removed some redundant info alerts

BEFORE

<img width="1012" alt="old" src="https://github.com/Altinn/altinn-studio/assets/24462611/2d9a6d16-55e4-496b-9d15-95873e9a1304">

AFTER

<img width="1010" alt="new" src="https://github.com/Altinn/altinn-studio/assets/24462611/d6b85b78-8e82-470a-ac94-2e206903a324">

nkylstad commented on frontend/language/src/nb.json:
```diff
@@ -49,44 +49,49 @@
-  "app_publish.private_app_owner_help": "For å publisere applikasjonen til et kjøretidsmiljø må den opprettes på vegne av en organisasjon. <a>Ta kontakt med oss</a> dersom du trenger hjelp.",
-  "app_publish.private_app_owner_info": "Altinn Studio er et verktøy som organisasjoner kan bruke for å lage og publisere applikasjoner. Du har nå åpnet en applikasjon som eies av en privatperson. Det betyr at du ikke kan publisere denne applikasjonen.",
-  "app_publish.private_app_owner_options": "Om du bare tester ut Altinn Studio vil du kunne teste ut all funksjonalitet vi har for å bygge den uten at du representerer en organisasjon. Du kan benytte forhåndsvisningen for å se hvordan et ferdig skjema ser ut.",
+  "app_deployment.btn_deploy_new_version": "Deploy ny versjon",
```
```suggestion
  "app_deployment.btn_deploy_new_version": "Publiser ny versjon",
```

nkylstad commented on frontend/language/src/nb.json:
```diff
@@ -49,44 +49,49 @@
-  "app_publish.private_app_owner_info": "Altinn Studio er et verktøy som organisasjoner kan bruke for å lage og publisere applikasjoner. Du har nå åpnet en applikasjon som eies av en privatperson. Det betyr at du ikke kan publisere denne applikasjonen.",
-  "app_publish.private_app_owner_options": "Om du bare tester ut Altinn Studio vil du kunne teste ut all funksjonalitet vi har for å bygge den uten at du representerer en organisasjon. Du kan benytte forhåndsvisningen for å se hvordan et ferdig skjema ser ut.",
+  "app_deployment.btn_deploy_new_version": "Deploy ny versjon",
+  "app_deployment.choose_version": "Velg ønsket versjon for deploy",
```
```suggestion
  "app_deployment.choose_version": "Velg versjon som skal publiseres",
```

nkylstad commented on frontend/language/src/nb.json:
```diff
@@ -49,44 +49,49 @@
-  "app_publish.private_app_owner_options": "Om du bare tester ut Altinn Studio vil du kunne teste ut all funksjonalitet vi har for å bygge den uten at du representerer en organisasjon. Du kan benytte forhåndsvisningen for å se hvordan et ferdig skjema ser ut.",
+  "app_deployment.btn_deploy_new_version": "Deploy ny versjon",
+  "app_deployment.choose_version": "Velg ønsket versjon for deploy",
+  "app_deployment.deploy_confirmation": "Er du sikker på at du vil deploye versjon {{selectedImageTag}} til miljøet? Dette vil overskrive eksisterende versjon {{appDeployedVersion}}.",
```
```suggestion
  "app_deployment.deploy_confirmation": "Er du sikker på at du vil publisere versjon {{selectedImageTag}} til miljøet? Dette vil overskrive eksisterende versjon {{appDeployedVersion}}.",
```

nkylstad commented on frontend/language/src/nb.json:
```diff
@@ -49,44 +49,49 @@
+  "app_deployment.btn_deploy_new_version": "Deploy ny versjon",
+  "app_deployment.choose_version": "Velg ønsket versjon for deploy",
+  "app_deployment.deploy_confirmation": "Er du sikker på at du vil deploye versjon {{selectedImageTag}} til miljøet? Dette vil overskrive eksisterende versjon {{appDeployedVersion}}.",
+  "app_deployment.deploy_confirmation_short": "Er du sikker på at du vil deploye versjon {{selectedImageTag}} til miljøet?",
```
```suggestion
  "app_deployment.deploy_confirmation_short": "Er du sikker på at du vil publisere versjon {{selectedImageTag}} til miljøet?",
```

nkylstad commented on frontend/language/src/nb.json:
```diff
@@ -49,44 +49,49 @@
+  "app_deployment.error": "Kunne ikke laste inn statusen og aktiviteten for denne applikasjonen. Prøv igjen senere.",
+  "app_deployment.last_published": "Sist publisert {{lastPublishedDate}}.",
+  "app_deployment.loading": "Laster inn statusen og aktiviteten for denne applikasjonen...",
+  "app_deployment.missing_rights": "Du har ikke rettigheter til å starte en deploy til {{envTitle}}. Tilgang kan delegeres av eiere i {{orgName}}.",
```
```suggestion
  "app_deployment.missing_rights": "Du har ikke rettigheter til å publisere til {{envTitle}}. Tilgang kan delegeres av eiere i {{orgName}}.",
```

nkylstad commented on frontend/language/src/nb.json:
```diff
@@ -49,44 +49,49 @@
+  "app_deployment.loading": "Laster inn statusen og aktiviteten for denne applikasjonen...",
+  "app_deployment.missing_rights": "Du har ikke rettigheter til å starte en deploy til {{envTitle}}. Tilgang kan delegeres av eiere i {{orgName}}.",
+  "app_deployment.no_env_1": "For å opprette miljøer, slik at du kan teste og produksjonssette må du ta kontakt med <0 href=\"mailto:tjenesteeier@altinn.no\">Altinn servicedesk</0>. Det er viktig at du oppgir organisasjonnavn og hvilke miljøer du ønsker tilgang til.",
+  "app_deployment.no_env_2": "Fram til et miljø er på plass, husk at du kan gjøre mye <0 href=\"https://docs.altinn.studio/app/testing/local/\">testing lokalt</0>.",
```
```suggestion
  "app_deployment.no_env_2": "Fram til et miljø er på plass, husk at du kan gjøre mye <0 href=\"https://docs.altinn.studio/app/testing/local/\">testing lokalt</0> eller i forhåndsvisningen i Altinn Studio.",
```

nkylstad commented on frontend/language/src/nb.json:
```diff
@@ -49,44 +49,49 @@
+  "app_deployment.no_env_2": "Fram til et miljø er på plass, husk at du kan gjøre mye <0 href=\"https://docs.altinn.studio/app/testing/local/\">testing lokalt</0>.",
+  "app_deployment.no_env_title": "Din organisasjon har ikke test- og produksjonsmiljø",
+  "app_deployment.no_team": "Du tilhører ikke noe team i organisasjonen din.",
+  "app_deployment.no_team_info": "For å få tilgang til å deploye applikasjonen din til et miljø er du nødt til å være lagt til i et team som har deploy rettigheter til det spesifikke miljøet.",
```
```suggestion
  "app_deployment.no_team_info": "For å få tilgang til å publisere applikasjonen din til et miljø er du nødt til å være lagt til i et team som har publiseringsrettigheter til det spesifikke miljøet.",
```

nkylstad commented on frontend/language/src/nb.json:
```diff
@@ -49,44 +49,49 @@
+  "app_deployment.no_env_title": "Din organisasjon har ikke test- og produksjonsmiljø",
+  "app_deployment.no_team": "Du tilhører ikke noe team i organisasjonen din.",
+  "app_deployment.no_team_info": "For å få tilgang til å deploye applikasjonen din til et miljø er du nødt til å være lagt til i et team som har deploy rettigheter til det spesifikke miljøet.",
+  "app_deployment.no_versions": "Det er ingen versjoner å deploye",
```
```suggestion
  "app_deployment.no_versions": "Det er ingen versjoner å publisere",
```

nkylstad commented on frontend/language/src/nb.json:
```diff
@@ -49,44 +49,49 @@
+  "app_deployment.no_team": "Du tilhører ikke noe team i organisasjonen din.",
+  "app_deployment.no_team_info": "For å få tilgang til å deploye applikasjonen din til et miljø er du nødt til å være lagt til i et team som har deploy rettigheter til det spesifikke miljøet.",
+  "app_deployment.no_versions": "Det er ingen versjoner å deploye",
+  "app_deployment.permission_checking": "Sjekker om du har tilgang til å legge ut appen",
```
```suggestion
  "app_deployment.permission_checking": "Sjekker om du har tilgang til å publisere ut applikasjonen",
```

nkylstad commented on frontend/language/src/nb.json:
```diff
@@ -49,44 +49,49 @@
+  "app_deployment.no_team_info": "For å få tilgang til å deploye applikasjonen din til et miljø er du nødt til å være lagt til i et team som har deploy rettigheter til det spesifikke miljøet.",
+  "app_deployment.no_versions": "Det er ingen versjoner å deploye",
+  "app_deployment.permission_checking": "Sjekker om du har tilgang til å legge ut appen",
+  "app_deployment.permission_error": "Kunne ikke sjekke om du har tilgang til å legge ut appen. Last siden på nytt for å prøve igjen.",
```
```suggestion
  "app_deployment.permission_error": "Kunne ikke sjekke om du har tilgang til å publisere applikasjonen. Last siden på nytt for å prøve igjen.",
```

nkylstad commented on frontend/language/src/nb.json:
```diff
@@ -49,44 +49,49 @@
+  "app_deployment.no_versions": "Det er ingen versjoner å deploye",
+  "app_deployment.permission_checking": "Sjekker om du har tilgang til å legge ut appen",
+  "app_deployment.permission_error": "Kunne ikke sjekke om du har tilgang til å legge ut appen. Last siden på nytt for å prøve igjen.",
+  "app_deployment.pipeline_deployment.build_result.canceled": "Vi opplever en feil og har derfor stoppet din deploy.",
```
```suggestion
  "app_deployment.pipeline_deployment.build_result.canceled": "Vi opplever en feil og har derfor stoppet din deploy.",
```
```suggestion
  "app_deployment.pipeline_deployment.build_result.canceled": "Vi opplever en feil og har derfor stoppet publisering av applikasjonen.",
```

nkylstad commented on frontend/language/src/nb.json:
```diff
@@ -49,44 +49,49 @@
+  "app_deployment.permission_checking": "Sjekker om du har tilgang til å legge ut appen",
+  "app_deployment.permission_error": "Kunne ikke sjekke om du har tilgang til å legge ut appen. Last siden på nytt for å prøve igjen.",
+  "app_deployment.pipeline_deployment.build_result.canceled": "Vi opplever en feil og har derfor stoppet din deploy.",
+  "app_deployment.pipeline_deployment.build_result.failed": "Noe gikk galt under deploy. Gjør en ny deploy eller forsøk igjen senere.",
```
```suggestion
  "app_deployment.pipeline_deployment.build_result.failed": "Noe gikk galt under publisering. Gjør en ny publisering eller forsøk igjen senere.",
```

nkylstad commented on frontend/language/src/nb.json:
```diff
@@ -49,44 +49,49 @@
+  "app_deployment.permission_error": "Kunne ikke sjekke om du har tilgang til å legge ut appen. Last siden på nytt for å prøve igjen.",
+  "app_deployment.pipeline_deployment.build_result.canceled": "Vi opplever en feil og har derfor stoppet din deploy.",
+  "app_deployment.pipeline_deployment.build_result.failed": "Noe gikk galt under deploy. Gjør en ny deploy eller forsøk igjen senere.",
+  "app_deployment.pipeline_deployment.build_result.none": "Deployment er satt i gang...",
```
```suggestion
  "app_deployment.pipeline_deployment.build_result.none": "Publisering er satt i gang...",
```

nkylstad commented on frontend/language/src/nb.json:
```diff
@@ -49,44 +49,49 @@
+  "app_deployment.pipeline_deployment.build_result.canceled": "Vi opplever en feil og har derfor stoppet din deploy.",
+  "app_deployment.pipeline_deployment.build_result.failed": "Noe gikk galt under deploy. Gjør en ny deploy eller forsøk igjen senere.",
+  "app_deployment.pipeline_deployment.build_result.none": "Deployment er satt i gang...",
+  "app_deployment.pipeline_deployment.build_result.partiallySucceeded": "Versjonen er deployet i miljøet, men inneholder feil / mangler.",
```
```suggestion
  "app_deployment.pipeline_deployment.build_result.partiallySucceeded": "Versjonen er publisert i miljøet, men inneholder feil / mangler.",
```

nkylstad commented on frontend/language/src/nb.json:
```diff
@@ -49,44 +49,49 @@
+  "app_deployment.pipeline_deployment.build_result.failed": "Noe gikk galt under deploy. Gjør en ny deploy eller forsøk igjen senere.",
+  "app_deployment.pipeline_deployment.build_result.none": "Deployment er satt i gang...",
+  "app_deployment.pipeline_deployment.build_result.partiallySucceeded": "Versjonen er deployet i miljøet, men inneholder feil / mangler.",
+  "app_deployment.pipeline_deployment.build_result.succeeded": "Versjonen er deployet i miljøet.",
```
```suggestion
  "app_deployment.pipeline_deployment.build_result.succeeded": "Versjonen er publisert i miljøet.",
```

nkylstad commented on frontend/language/src/nb.json:
```diff
@@ -49,44 +49,49 @@
+  "app_deployment.table.build_log": "Byggloggen",
+  "app_deployment.table.build_log_active_link": "Se logg",
+  "app_deployment.table.build_log_expired_link": "Utløpt",
+  "app_deployment.table.deployed_by_col": "Deployet av",
```
```suggestion
  "app_deployment.table.deployed_by_col": "Publisert av",
```

nkylstad commented on frontend/language/src/nb.json:
```diff
@@ -49,44 +49,49 @@
+  "app_deployment.table.build_log_active_link": "Se logg",
+  "app_deployment.table.build_log_expired_link": "Utløpt",
+  "app_deployment.table.deployed_by_col": "Deployet av",
+  "app_deployment.table.deployed_version_history": "Versjoner som er deployet til {{envTitle}}",
```
```suggestion
  "app_deployment.table.deployed_version_history": "Versjoner som er publisert til {{envTitle}}",
```

nkylstad commented on frontend/language/src/nb.json:
```diff
@@ -49,44 +49,49 @@
+  "app_deployment.table.build_log_expired_link": "Utløpt",
+  "app_deployment.table.deployed_by_col": "Deployet av",
+  "app_deployment.table.deployed_version_history": "Versjoner som er deployet til {{envTitle}}",
+  "app_deployment.table.deployed_version_history_empty": "Det er ingen deploy historikk i {{envTitle}}",
```
```suggestion
  "app_deployment.table.deployed_version_history_empty": "Det er ingen publiseringshistorikk i {{envTitle}}",
```

nkylstad commented on frontend/language/src/nb.json:
```diff
@@ -49,44 +49,49 @@
+  "app_deployment.table.deployed_version_history_empty": "Det er ingen deploy historikk i {{envTitle}}",
+  "app_deployment.table.status": "Status",
+  "app_deployment.table.version_col": "Versjon",
+  "app_deployment.technical_error_1": "Huff da, vi opplever en teknisk feil og får derfor ikke deployet applikasjonen din. Gjør en ny deploy eller forsøk igjen senere. Dersom problemet vedvarer, <a>ta kontakt med oss</a>.",
```
```suggestion
  "app_deployment.technical_error_1": "Huff da, vi opplever en teknisk feil og får derfor ikke publisert applikasjonen din. Gjør en ny deploy eller forsøk igjen senere. Dersom problemet vedvarer, <a>ta kontakt med oss</a>.",
```

nkylstad commented on frontend/language/src/nb.json:
```diff
@@ -629,26 +637,18 @@
-  "overview.loading_deploys": "Laster inn publiseringer",
-  "overview.loading_env": "Laster inn miljøer",
-  "overview.loading_page": "Laster inn oversiktssiden",
+  "overview.header_loading": "Laster inn navnet på appen",
```
```suggestion
  "overview.header_loading": "Laster inn navnet på applikasjonen",
```

nkylstad reviewed (changes requested):
Fantastic job 🥳 
I have added some suggestions to the different texts to try and make them more consistent (we use the term "deploy" and "publish" and should try to stick to one - I think publish is the most understandable as deploy is very technical). 

I think we should also create an issue to go over all the texts on the deploy page to see if there are any more changes that we should make to make the page easier to use, since my suggestions only change a few words  😊
